features/step_definitions/auditing_steps.rb

```diff
@@ -3,15 +3,24 @@
 Then("there is an audit entry containing {string}") do |content|
   step "I visit the audit log"
 
-  within_table("Activity log") do
+  within_table("Audit log") do
     expect(page).to have_content content
   end
 end
 
+Then("there is an audit entry containing {string} and {string}") do |content, value|
+  step "I visit the audit log"
+
+  steps %(
+    Then the page contains "#{content}"
+    And the page contains "#{value}"
+  )
+end
```

Comment:
I'll try the approach of searching within a table for the step to be more precise - I can't add two different lines within a single "And there is xx" because of the \n separators between the two lines (plus it'd look hellufmessy saying "Application returned \/n\ Applicant blah"

Reply:
I'd rather you have the copy merged together because it ties them together, otherwise the above case (where two different entries have the separate pieces of copy - whereas it should be within one) still pases.

Reply:
ohh wait I've found something saying to use triple quotes: https://stackoverflow.com/questions/35639150/is-it-possible-to-write-a-gherkin-step-on-multiple-lines
gonna give it a go

Reply:
good shout on the triple quotes - I couldn't get them to work, but don't know if you had any more luck? Here's a solution that I actually am finding nicer, because it removes the whole `\n` awkwardness; it uses the `normalize_ws` option of Capybara that turns these pesky newlines into normal spaces: https://rubydoc.info/github/jnicklas/capybara/Capybara%2FNode%2FMatchers:assert_text


```patch
From 74d5a0aea54f36d299d1d657b67709847cc65018 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?St=C3=A9phane=20Maniaci?=
 <stephane.maniaci@unboxedconsulting.com>
Date: Fri, 10 Dec 2021 11:19:51 +0000
Subject: [PATCH] audit: simplify the feature tests

---
 features/auditing.feature                   |  8 ++++----
 features/step_definitions/auditing_steps.rb | 11 +----------
 2 files changed, 5 insertions(+), 14 deletions(-)

diff --git a/features/auditing.feature b/features/auditing.feature
index 46fac3b2..22701a87 100644
--- a/features/auditing.feature
+++ b/features/auditing.feature
@@ -43,9 +43,9 @@ Feature: Auditing a planning application
     And I check "Broads"
     And I uncheck "Conservation Area"
     And I press "Save"
-    Then there is an audit entry containing "Constraint added" and "National Park"
-    And there is an audit entry containing "Constraint added" and "Broads"
-    And there is an audit entry containing "Constraint removed" and "Conservation Area"
+    Then there is an audit entry containing "Constraint added National Park"
+    And there is an audit entry containing "Constraint added Broads"
+    And there is an audit entry containing "Constraint removed Conservation Area"
 
   Scenario: I can view an entry in the audit log showing application updates
     Given I press "Application information"
@@ -55,7 +55,7 @@ Feature: Auditing a planning application
     Then there is an audit entry containing "Address 1 updated"
     And there is an audit entry containing "Miyazaki"
     And there is an audit entry containing "Changed to: 20 leafy gardens"
-    
+
   Scenario: I can view an entry in the audit showing the application being validated
     When I press "Validate application"
     And I press "Mark the application as valid"
diff --git a/features/step_definitions/auditing_steps.rb b/features/step_definitions/auditing_steps.rb
index 890b0b35..982d775a 100644
--- a/features/step_definitions/auditing_steps.rb
+++ b/features/step_definitions/auditing_steps.rb
@@ -4,16 +4,7 @@ Then("there is an audit entry containing {string}") do |content|
   step "I visit the audit log"
 
   within_table("Audit log") do
-    expect(page).to have_content content
-  end
-end
-
-Then("there is an audit entry containing {string} and {string}") do |content, value|
-  step "I visit the audit log"
-
-  within_table("Audit log") do
-    expect(page).to have_content content
-    expect(page).to have_content value
+    expect(page).to have_content(content, normalize_ws: true)
   end
 end
 
-- 
2.33.1

```

Reply:
they weren't working for me either so I went with the variation you suggested, cheers!